### DISCUSSION TITLE
getting name instead of user id

### DISCUSSION THREAD

**Discussion by mehradamiri** [pocketbase/pocketbase#3004] [Q&A] [answered]
i have user table that has campagins field and it has relation with 'campaign' table and in 'campagins' table i have field named 'dm' and that has relation with 'users' table, in profile page i will get data like this 

```js
 const needsfields = [
    "id",
    "username",
    "email",
    "name",
    "avatar",
    "collectionId",
    "bio",
    "playType",
    "blogs",
    "companions",
    "expand.blogs.name",
    "expand.blogs.discription",

    "expand.blogs.collectionId",
    "expand.blogs.id",
    "expand.blogs.cover",
    "expand.blogs.tags",
    "campagins",
    "expand.campagins.id",
    "expand.campagins.name",
    "expand.campagins.dm",
    "expand.campagins.isEnd",
    "expand.campagins.cover",
    "expand.campagins.collectionId",
  ];
  const formatedFieldString = needsfields.toString();

  const data = serializeNonPOJOs(
    await locals.pb.collection("users").getFullList({
      filter: `username='${username}'`,
      expand: "blogs,campagins",
      fields: formatedFieldString,
    })
  );
```

in 'expand.campagins.dm' i only get 'dm' id how can i acsses username of 'dm'

**Comment by ganigeorgiev** [comment]
If `dm` is a relation of `campagins`, then you need to also add it with dot-notation in the `expand` field (aka. `expand: "blogs,campagins.dm"`).

The `fields` value should correspond to the json response structure, aka. for nested expands, it will should be `expand.campagins.expand.dm.name`.

_On a side-note, if you are returning so many fields anyway it may not be worth filtering them with `fields`._

**Reply by mehradamiri** [reply]
i use fields becuse in blog table I have field named main_text and it maight be long text that I don't needed

**Reply by mehradamiri** [reply]
![image](https://github.com/pocketbase/pocketbase/assets/92649809/3ded46b4-2f6d-4401-a112-7e211e70c642)
i still don't get dm name

**Reply by ganigeorgiev** [reply, marked as answer]
The `dm` doesn't seem to be expanded. Make sure that your client request has View API rule access to the `dm` related collection.

**Reply by mehradamiri** [reply]
thank you for mentioned that information the reason was in users API rules in view rule field that has this default rule  
```
id = @request.auth.id
```
can you tell me a reason of this rule ?

> The `dm` doesn't seem to be expanded. Make sure that your client request has View API rule access to the `dm` related collection.

**Reply by ganigeorgiev** [reply]
> can you tell me a reason of this rule ?

I guess `dm` points to the default `users` auth collection. The above rule is set by default to restrict only the authenticated users to have view access to their user record (aka. `getOne()`).

You are free to change it to whatever you want.
